xapian-letor/feature/colltfcolllenfeature.cc

```diff
@@ -38,37 +38,74 @@ CollTfCollLenFeature::name() const
 vector<double>
 CollTfCollLenFeature::get_values() const
 {
-    LOGCALL(API, std::vector<double>, "CollTfCollLenFeature::get_values", NO_ARGS);
+    LOGCALL(API, vector<double>, "CollTfCollLenFeature::get_values", NO_ARGS);
 
     vector<double> values;
     double value = 0;
+    double coll_len = 0;
+    map<string, Xapian::termcount>::const_iterator coll_len_iterator =
+	    collection_length.find("title");
+    if (coll_len_iterator != collection_length.end())
+	coll_len = (double)coll_len_iterator->second;
+    else
+	coll_len = 0;
```

Comment:
You already set `coll_len` to 0 above.  This pattern seems (a) redundant and (b) liable to trigger "value is assigned bu not used" compiler warnings.  It's repeated many times below too.

Given in some of those places you reuse the variable assigned in this way (which I'm not madly keen on, but let's ignore that for now) I would drop assignment in the declaration, so the consistent pattern is the if/else assignment.